src/Docker/Runner/DataLoader/DataLoader.php

```diff
@@ -138,15 +144,23 @@ public function __construct(
         $tokenInfo = $this->clientWrapper->getBasicClient()->verifyToken();
 
         /* dataDirectory is "something/data" - this https://github.com/keboola/docker-bundle/blob/f9d4cf0d0225097ba4e5a1952812c405e333ce72/src/Docker/Runner/WorkingDirectory.php#L90
-            we need need the base dir here */
+            we need the base dir here */
         $dataDirectory = dirname($this->dataDirectory);
 
-        $workspaceProviderFactory = new ComponentWorkspaceProviderFactory(
-            new Components($this->clientWrapper->getBasicClient()),
-            new Workspaces($this->clientWrapper->getBasicClient()),
-            $this->component->getId(),
-            $this->configId
-        );
+        /* There can only be one workspace type (ensured in validateStagingSetting()) - so we're checking
```

Comment:
Nebylo by lepsi tohle vythanout do samostatny tridy, metodu, ktera vraci patricnou varianty `WorkspaceProviderFactoryInterface`. Myslim ze by to i pekne zjednodusilo testy.

Reply:
Switching to oraculum mode.. jako neni to nejhezci, ale samozrejme na prvni dobrou se nabizi `WorkspaceProviderFactoryFactory`

Reply:
toho jsem se bal 😆

Reply:
No ale mohl by si to ted rozdelit do dvou "jednodussich" testu, ne? Myslim udelat
* separatni `WorkspaceProviderFactoryFactory`, ktery bude resit, jestli ta factory vraci spravny typ/spravne nakonfigurovany `workspaceProviderFactory`
* separatni test na `DataLoader`, ktery bude resit jen, ze se ta vyrobena factory pouzije a to u zmuze byt jen jednou (nemusis testovat vsechny typy factory) nebo dokonce mozna s nejakym mockem

Reply:
K tomu potřebuju otestovat, že nejen, že se to správně nakonfiguruje, ale hlavně to, že se ten workspace nesmaže nebo nevytvoří dvakrát (čili ta druhá odrážka mě nepomůže)